edx_proctoring/static/proctoring/js/views/proctored_exam_view.js

```diff
@@ -71,6 +76,22 @@ var edx = edx || {};
                 "credit eligibility in this course.\n");
         },
         updateRemainingTime: function (self) {
+            self.timerTick ++;
+            if (self.timerTick % 5 == 0){
+                var url = self.model.url + '/' + self.model.get('attempt_id');
+                $.ajax(url).success(function(data) {
+                    if (data.status === 'error') {
+                        // Let the student know that his exam has failed due to an error.
+                        // This alert may or may not bring the browser window back to the
+                        // foreground (depending on browser as well as user settings)
+                        alert(gettext('Your exam has failed'));
```

Comment:
Submitted in a proctoring exam are always recorded as the student takes the test (just like a regular exam). Although - that said - we might be messaging things inconsistently regarding that.

Regarding sessions, that's a bit more complicated. What I *think* the answer is:

1) If user closes SoftwareSecure software, then the session is uploaded to SoftwareSecure for review
2) If the software crashes or there is some other abnormal termination (e.g. power loss, restarting machine), then I'd assume that everything is lost.

However, we probably need some feedback from product on this one. I think in either case, the exam is considered "invalid", but I think that is for them to confirm.